libnetwork/internal/resolvconf/resolvconf_unix_test.go

```diff
@@ -0,0 +1,817 @@
+			inputSearch:  []string{"invalid"},
+			inputOptions: []string{"ndots:0"},
+			expContent: `nameserver 1.2.3.4
+search invalid
+options ndots:0
+
+# Based on host file: ''
+# Overrides: []
+# Option ndots from: host
+`,
+		},
+		{
+			name:         "Empty overrides",
+			inputNS:      []string{"1.2.3.4"},
+			inputSearch:  []string{"invalid"},
+			inputOptions: []string{"ndots:0"},
+			expContent: `
+# Based on host file: ''
+# Overrides: [nameservers search options]
+`,
+		},
+		{
+			name:            "Overrides",
+			inputNS:         []string{"1.2.3.4"},
+			inputSearch:     []string{"invalid"},
+			inputOptions:    []string{"ndots:0"},
+			overrideNS:      []string{"2.3.4.5", "fdba:acdd:587c::53"},
+			overrideSearch:  []string{"com", "invalid", "example"},
+			overrideOptions: []string{"ndots:1", "edns0", "trust-ad"},
+			expContent: `nameserver 2.3.4.5
+nameserver fdba:acdd:587c::53
+search com invalid example
+options ndots:1 edns0 trust-ad
+
+# Based on host file: ''
+# Overrides: [nameservers search options]
+# Option ndots from: override
+`,
+		},
+		{
+			name:         "Add option no overrides",
+			noOverrides:  true,
+			inputNS:      []string{"1.2.3.4"},
+			inputSearch:  []string{"invalid"},
+			inputOptions: []string{"ndots:0"},
+			addOption:    "attempts:3",
+			expContent: `nameserver 1.2.3.4
+search invalid
+options ndots:0 attempts:3
+
+# Based on host file: ''
+# Overrides: []
+# Option ndots from: host
+`,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
+			var input string
+			if len(tc.inputNS) != 0 {
+				for _, ns := range tc.inputNS {
+					input += "nameserver " + ns + "\n"
+				}
+			}
+			if len(tc.inputSearch) != 0 {
+				input += "search " + strings.Join(tc.inputSearch, " ") + "\n"
+			}
+			if len(tc.inputOptions) != 0 {
+				input += "options " + strings.Join(tc.inputOptions, " ") + "\n"
+			}
+			rc, err := Parse("", bytes.NewBuffer([]byte(input)))
+			assert.NilError(t, err)
+			assert.Check(t, is.DeepEqual(a2s(rc.NameServers()), tc.inputNS))
+			assert.Check(t, is.DeepEqual(rc.Search(), tc.inputSearch))
+			assert.Check(t, is.DeepEqual(rc.Options(), tc.inputOptions))
+
+			if !tc.noOverrides {
+				rc.OverrideNameServers(tc.overrideNS)
+				rc.OverrideSearch(tc.overrideSearch)
+				rc.OverrideOptions(tc.overrideOptions)
+
+				assert.Check(t, is.DeepEqual(a2s(rc.NameServers()), tc.overrideNS))
+				assert.Check(t, is.DeepEqual(rc.Search(), tc.overrideSearch))
+				assert.Check(t, is.DeepEqual(rc.Options(), tc.overrideOptions))
+			}
+
+			if tc.addOption != "" {
+				options := rc.Options()
+				rc.AddOption(tc.addOption)
+				assert.Check(t, is.DeepEqual(rc.Options(), append(options, tc.addOption)))
+			}
+
+			d := t.TempDir()
+			path := filepath.Join(d, "resolv.conf")
+			err = rc.WriteFile(path, "", 0644)
+			assert.NilError(t, err)
+
+			content, err := os.ReadFile(path)
+			assert.NilError(t, err)
+			// fmt.Print(string(content))
```

Comment:
That's useful; don't comment it out!
```suggestion
			t.Logf("%s", content)
```

Reply:
It was useful for grabbing the expected output when writing the test - but it'll show up in the DeepEqual output if the test fails ... done anyway.